### ISSUE TITLE
ArgumentOutOfRangeException for Aggregation with PeriodSize.Month

### ISSUE THREAD

**Issue by SvenSkrabal** [DaveSkender/Stock.Indicators#727]
### the problem

Calling `public static IEnumerable<Quote> Aggregate<TQuote>(this IEnumerable<TQuote> quotes, PeriodSize newSize)` with `PeriodSize.Month` returns a `ArgumentOutOfRangeException`. Aggregation by other periods in lower time frame (Daily, Weekly) works as expected.

### to reproduce

1. Create a `IEnumerable<Quote> quotes` collection with hourly or daily data
2. Call `quotes.Aggregate(PeriodSize.Month)`

Error message(s):

```
System.ArgumentOutOfRangeException
  HResult=0x80131502
  Message=Historical quotes Aggregation must use a new size value that is greater than zero (0). Arg_ParamName_Name
ArgumentOutOfRange_ActualValue
  Source=Skender.Stock.Indicators
  StackTrace:
   at Skender.Stock.Indicators.HistoricalQuotes.Aggregate[TQuote](IEnumerable`1 quotes, TimeSpan timeSpan)
   at Skender.Stock.Indicators.HistoricalQuotes.Aggregate[TQuote](IEnumerable`1 quotes, PeriodSize newSize)
```

**Comment by DaveSkender** [comment]
Hmm, interesting.  This exception should only be thrown when the equivalent TimeSpan is less than or equal to zero.  Will have to take a closer look, maybe add some more unit tests to check for conversion issues between PeriodSize and TimeSpan.

**Comment by SvenSkrabal** [comment]
The `PeriodSize` enum value is converted to a `TimeSpan` object with an extension method:
```
public static IEnumerable<Quote> Aggregate<TQuote>(
        this IEnumerable<TQuote> quotes,
        PeriodSize newSize)
        where TQuote : IQuote
    {
        // parameter conversion
        TimeSpan newTimeSpan = newSize.ToTimeSpan();

        // convert
        return quotes.Aggregate(newTimeSpan);
    }
```

This extension method is lacking a case for PeriodSize.Month:
```
// PERIOD-SIZE to TIMESPAN CONVERSION
    internal static TimeSpan ToTimeSpan(this PeriodSize periodSize)
    {
        return periodSize switch
        {
            PeriodSize.OneMinute => TimeSpan.FromMinutes(1),
            PeriodSize.TwoMinutes => TimeSpan.FromMinutes(2),
            PeriodSize.ThreeMinutes => TimeSpan.FromMinutes(3),
            PeriodSize.FiveMinutes => TimeSpan.FromMinutes(5),
            PeriodSize.FifteenMinutes => TimeSpan.FromMinutes(15),
            PeriodSize.ThirtyMinutes => TimeSpan.FromMinutes(30),
            PeriodSize.OneHour => TimeSpan.FromHours(1),
            PeriodSize.TwoHours => TimeSpan.FromHours(2),
            PeriodSize.FourHours => TimeSpan.FromHours(4),
            PeriodSize.Day => TimeSpan.FromDays(1),
            PeriodSize.Week => TimeSpan.FromDays(7),
            _ => TimeSpan.Zero
        };
    }
```

**Comment by SvenSkrabal** [comment]
I guess that this behavior is intentional due to the fact that every month has a different number of days. Probably you just cannot round down the dates as with other periods that are evenly spaced. This is just an assumption and hasnt been tested or verified by me in any form.

**Comment by DaveSkender** [comment]
I think you’re right, and thanks for troubleshooting.  `TimeSpan`, being an `enum` itself, would have to be a fixed time unit, and months are not fixed sizes.

With that said, if using the `PeriodSize` overload variant of `Aggregate()`, we could probably implement a special case for `Month` where it rolls up on unique year and month combo number.

**Comment by DaveSkender** [comment]
In this PR, I'm essentially adding a custom path for rolling up by month.  Let me know what you think.

```csharp
return quotes
  .OrderBy(x => x.Date)
  .GroupBy(x => new DateTime(x.Date.Year, x.Date.Month, 1))
  .Select(x => new Quote
  {
      Date = x.Key,
      Open = x.First().Open,
      High = x.Max(t => t.High),
      Low = x.Min(t => t.Low),
      Close = x.Last().Close,
      Volume = x.Sum(t => t.Volume)
  });
```

**Comment by SvenSkrabal** [comment]
Looks great. I thought of something similar. Thanks for fast fixing 👍

**Comment by DaveSkender** [comment]
This was implemented in version 1.22.0 and is now available.
